### PR TITLE
Adds more error cases in hit the point functionality

### PR DESCRIPTION
**Pull request by OctavioBR**
Guys, after a long discussion, we decided to keep developing this app. Here goes some more tests that we managed to code during the last half of our meeting :sweat_smile:

### CODE REVIEW

**Comment by fernandoguedes** [comment]
Nice!

File: client/karma.conf.js [hunk 1]
```diff
@@ -25,10 +25,10 @@ module.exports = function (config) {
...
 
     coverageReporter: {
-        reporters: [
-            // generates ./coverage/coverage-final.json
-            {type:'json', subdir: '.'},
-        ]
+      reporters: [
+          // generates ./coverage/coverage-final.json
+          {type: 'json', subdir: '.'}
+      ]
```
Comment (vitorarins) [inline client/karma.conf.js]: Thanks 😅

File: client/src/clock-register/clock-register.controller.js [hunk 2]
```diff
@@ -44,6 +44,13 @@
                 break;
               }
 
+              case 500: {
+                if (error.data.message) {
```
Comment (vitorarins) [inline client/src/clock-register/clock-register.controller.js]: I think this handling should be done in the first `if` on line 31. Also it looks like an inconsistency with `error.message`.

File: client/src/clock-register/clock-register.directive.spec.js [hunk 3]
```diff
@@ -44,20 +46,43 @@ describe('dojo clock register directive', function () {
...
+          firstItem = element.find('#daily-list :eq(0)');
 
-        expect(firstItem.text()).toBe('16:20:00');
-      });
+          expect(firstItem.text()).toBe('16:20:00');
+        });
+
+        it('if session is lost, should warn the user with message', function () {
+          var feedbackMessage;
+          $httpBackend.expect('POST', '/api/v1/clock/hit').respond(401, 'Authentication Failed');
```
Comment (vitorarins) [inline client/src/clock-register/clock-register.directive.spec.js]: I think this should return an object like the following test.

**Review by vitorarins — CHANGES_REQUESTED** [review]
Looks ok. Just have some doubts about the error handling 😉

File: client/src/clock-register/clock-register.controller.js (hunk 2 above)
Comment (OctavioBR) [inline client/src/clock-register/clock-register.controller.js]: Absolutely. We completely forgot about that `if` in the beginning of `catch` callback. As far as I remember, we started discussing about "error 500" and using a "convention" to only deal with more specific body formats (like objects that contains a certain key), then we did this. But I agree with you that we should be more concise.

File: client/src/clock-register/clock-register.directive.spec.js (hunk 3 above)
Comment (OctavioBR) [inline client/src/clock-register/clock-register.directive.spec.js]: Yes! I think we should stick to objects or only direct body strings. Not mixing both!

**Review by vitorarins — APPROVED** [review]

**Comment by OctavioBR** [comment]
Abandoned 💀

**Pull request closed without merging**